### PR TITLE
feat(profiling): onboarding boilerplate for profiling

### PR DESCRIPTION
**Pull request by JonasBa**
Renames the old onboarding to legacy and introduces the boilerplate for the new one. There will be a few PR's to follow for each step and we'll make the switch and remove the legacy onboarding.

### CODE REVIEW

File: static/app/routes.tsx [hunk 1]
```diff
@@ -1688,7 +1688,7 @@ function buildRoutes() {
       <IndexRoute component={make(() => import('sentry/views/profiling/content'))} />
       <Route
         path="onboarding/"
-        component={make(() => import('sentry/views/profiling/onboarding'))}
+        component={make(() => import('sentry/views/profiling/legacyOnboarding'))}
```
Comment (Zylphrex) [inline static/app/routes.tsx:1691]: This going to break the current setup onboarding button temporarily. Let's not rename it until the new one is ready.

File: static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    if (nextStep === null) {
+      return;
+    }
+
+    setState(current => {
+      const next = {...nextStep, next: null, previous: current};
+      // Add the edges between the old and the new step
+      current.next = next;
+      next.previous = current;
+      // Return the neext step
```
Comment (Zylphrex) [inline static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx:30]:
```suggestion
      // Return the next step
```

File: static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            : IOSSendDebugFilesInstruction,
+        next: null,
+      });
+    },
+    [step, toStep]
+  );
+
+  return (
+    <ModalBody>
+      <ModalHeader>Select a Project</ModalHeader>
```
Comment (Zylphrex) [inline static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx:79]:
```suggestion
      <ModalHeader>{t('Select a Project')}</ModalHeader>
```

File: static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+function AndroidSendDebugFilesInstruction({
+  Body: ModalBody,
+  Header: ModalHeader,
+  Footer: ModalFooter,
+  toStep,
+  step,
+}: OnboardingStepProps) {
+  return (
+    <ModalBody>
+      <ModalHeader>Send Debug Files For Android</ModalHeader>
```
Comment (Zylphrex) [inline static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx:99]:
```suggestion
      <ModalHeader>{t('Send Debug Files For Android')}</ModalHeader>
```

File: static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+  Body: ModalBody,
+  Header: ModalHeader,
+  Footer: ModalFooter,
+  toStep,
+  step,
+}: OnboardingStepProps) {
+  // Required as typescript cannot properly infer the previous step
+  return (
+    <ModalBody>
+      <ModalHeader>Send Debug Files For iOS</ModalHeader>
```
Comment (Zylphrex) [inline static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx:121]:
```suggestion
      <ModalHeader>{t('Send Debug Files For iOS')}</ModalHeader>
```

File: static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx (hunk 2 above)
Comment (JonasBa) [reply static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx:30]: I fixed this and the followup comments below. I only kept them as placeholders, they should be all t'd up in https://github.com/getsentry/sentry/pull/37118/files and I also changed the content to what we had in the designs.

File: static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx (hunk 3 above)
Comment (JonasBa) [reply static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx:79]: Same comment as https://github.com/getsentry/sentry/pull/37068#discussion_r931308944

File: static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx (hunk 4 above)
Comment (JonasBa) [reply static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx:99]: Same as https://github.com/getsentry/sentry/pull/37068#discussion_r931308944

File: static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx (hunk 5 above)
Comment (JonasBa) [reply static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx:121]: Same https://github.com/getsentry/sentry/pull/37068#discussion_r931308944

File: static/app/routes.tsx (hunk 1 above)
Comment (JonasBa) [reply static/app/routes.tsx:1691]: Why would that be? Did I forget to update something? It should still be the /onboarding route, but just load the different file

Comment (Zylphrex) [reply static/app/routes.tsx:1691]: mistook it for the url, this looks good 👍

**Review by Zylphrex — APPROVED** [review]

**Pull request merged**